driver-scala/src/main/scala/org/mongodb/scala/ObservableImplicits.scala

```diff
@@ -116,17 +118,41 @@ trait ObservableImplicits {
     override def subscribe(observer: Observer[_ >: GridFSFile]): Unit = Mono.from(publisher).subscribe(observer)
   }
 
-  implicit class ToSingleObservableVoid(pub: => Publisher[Void]) extends SingleObservable[Void] {
+  /**
+   * A `GridFSUploadPublisher`` that emits
+   *
+   *   - exactly one item, if the wrapped `Publisher` does not signal an error, even if the represented stream is empty;
+   *   - no items if the wrapped `Publisher` signals an error.
+   *
+   * @param pub A `Publisher` representing a finite stream.
+   */
+  implicit class ToGridFSUploadPublisherUnit(pub: => GridFSUploadPublisher[Void]) extends GridFSUploadPublisher[Unit] {
     val publisher = pub
-    override def subscribe(observer: Observer[_ >: Void]): Unit =
-      Mono
-        .from(pub)
-        .subscribe(
-          (_: Void) => {},
-          (e: Throwable) => observer.onError(e),
-          () => observer.onComplete(),
-          (s: JSubscription) => observer.onSubscribe(s)
-        )
+
+    override def subscribe(observer: Subscriber[_ >: Unit]): Unit =
+      Flux.from(publisher).reduce((), (_: Unit, _: Void) => ()).subscribe(observer)
+
+    override def getObjectId: ObjectId = publisher.getObjectId
+
+    override def getId: BsonValue = publisher.getId
+  }
+
+  /**
+   * An [[Observable]] that emits
+   *
+   *   - exactly one item, if the wrapped `Publisher` does not signal an error, even if the represented stream is empty;
+   *   - no items if the wrapped `Publisher` signals an error.
+   *
+   * @param pub A `Publisher` representing a finite stream.
+   */
+  implicit class ToSingleObservableUnit(pub: => Publisher[Void]) extends SingleObservable[Unit] {
```

Comment:
This class is why we are able to simply replace `Observable[Void]` with `Observable[Unit]` in the codebase.

Reply:
implicits are magic and have been changed in Scala 3 in favour of extension methods (probably more akin to Kotlin extension methods).

Reply:
yeap 👍